src/sentry/replays/consumers/recording.py

```diff
@@ -102,9 +105,22 @@ def initialize_message_context(message: Message[KafkaPayload]) -> MessageContext
     return MessageContext(message_dict, transaction, current_hub)
 
 
-def move_replay_to_permanent_storage(message: Message[MessageContext]) -> Any:
+def process_message_threaded(message: Message[MessageContext]) -> Any:
     """Move the replay payload to permanent storage."""
     context: MessageContext = message.payload
     message_dict = context.message
 
     ingest_recording(message_dict, context.transaction, context.current_hub)
+
+
+def process_message(message: Message[KafkaPayload]) -> Any:
+    """Move the replay payload to permanent storage."""
+    transaction = sentry_sdk.start_transaction(
+        name="replays.consumer.process_recording",
+        op="replays.consumer",
+        sampled=random.random()
+        < getattr(settings, "SENTRY_REPLAY_RECORDINGS_CONSUMER_APM_SAMPLING", 0),
+    )
+    current_hub = sentry_sdk.Hub(sentry_sdk.Hub.current)
+    message_dict = RECORDINGS_CODEC.decode(message.payload.value)
+    ingest_recording(message_dict, transaction, current_hub)
```

Comment:
is it worth having a span for decoding and another for `ingest_recording`? Or is this happening somehow automatically?

also noob here but how is `transaction` finishing? It does so without explicitly `transaction.finish()` nor `with` ?

Reply:
Oh I see, it'll be closed within `ingest_recording`

Reply:
@bruno-garcia  yeah thats a fair point we should track the performance of this decoding step as well.